### ISSUE TITLE
Loss stop decline when train own LM model

### ISSUE THREAD

**Issue by RichardHWD** [zalandoresearch/flair#928]
I have Non-Latin corpus contain about 1,680,000 sentencs, and I follow Tutorial 9 to train my own LM model . I get my own char_dictionary by tutorial and set character_level = False (because it is Non-Latin corpus). 
The loss was 9.2 , lr = 20 when the trainning began. But now it is 3.17 lr is nearly 0 , ppl = 23.74 for a whole day.
I don`t know how to decline the loss , or it is the best ? 
Thanks for help!

**Comment by alanakbik** [comment]
Hello @RichardHWD if the loss is declining to quickly, you could increase the `patience` parameter. This parameter indicates how many training splits of no improvement are tolerated before decreasing the learning rate. 

BTW: If you've computed a character dictionary following the tutorial, you should set `character_level = True`. Otherwise, with a character dictionary the language model will not be able to learn much. Even if you have a non-latin language like Chinese or Japanese, a character dictionary is probably the best way to go since most complex expressions in such languages consist of several symbols.

**Comment by RichardHWD** [comment]
You're so efficient ! Thanks to your reply , I find out the problem in the "character_level" and dictionary. I will try again.
Your coding is beautiful !

**Comment by RichardHWD** [comment]
BTW: ppl = 23.74 is good or bad ....

**Comment by stefan-it** [comment]
@RichardHWD For a word-based language model this is totally fine. E.g. recent approaches like the Transformer-XL would achieve ppls around 21.8 on a one billion word benchmark (for English).

So your ppl looks reasonable :)
